src/Field/Email.php

```diff
@@ -84,14 +88,31 @@ public function normalize($value)
         return parent::normalize(implode(', ', $emails));
     }
 
+    /**
+     * Validate DNS.
+     */
+    protected function isDNSValid(string $domain): bool
+    {
+        return $this->hasDNSRecord($domain, true) || $this->hasDNSRecord($domain, false);
+    }
+
+    private function hasDNSRecord(string $domain, bool $isMX): bool
+    {
+        $normalizedDomain = $domain . '.';
+        if (!checkdnsrr($normalizedDomain, ($isMX ? 'MX' : 'A'))) {
+            return false;
+        }
+
+        // dns_get_record can also return false
+        $records = dns_get_record($normalizedDomain, ($isMX ? DNS_MX : DNS_A)) ?: [];
+
+        return !empty($records);
```

Comment:
much better to test `!== false && count() > 0`, then you are do not need any comment ;-)

Reply:
addressed